### DISCUSSION TITLE
how to publish or edit the package core files ex : withData file

### DISCUSSION THREAD

**Discussion by TheOWler** [rappasoft/laravel-livewire-tables#977] [Q&A]
hi, i want to do a custom query inside the executeQuery function located in vendor\rappasoft\laravel-livewire-tables\src\Traits\withData.php
but if i do in the futute the composer update it will be erased so any tips ?

**Comment by lrljoe** [comment]
What are you looking to achieve, as there is likely a better way forward?

**Reply by TheOWler** [reply]
basicaly i want to do a cursor pagination not only a pagination so when i dig in the code the only way i think it will work is by chaging the core files which is in this case the withData.php file 
i have succeed with this part and its working, the problem is if we run composer update in the futur it will get a fresh version of rappasoft package and the changes will be erased and then i need to get back to the code and upload the changes again, which can lead in the futur if rappasoft tables gets updated to problems with version mistmatch

**Reply by lrljoe** [reply]
> basicaly i want to do a cursor pagination not only a pagination so when i dig in the code the only way i think it will work is by chaging the core files which is in this case the withData.php file 
> i have succeed with this part and its working, the problem is if we run composer update in the futur it will get a fresh version of rappasoft package and the changes will be erased and then i need to get back to the code and upload the changes again, which can lead in the futur if rappasoft tables gets updated to problems with version mistmatch

Your best bet would be to take your modified files, extend them off the core files and then call your modified files instead .
